client-tools/Shell/ProcessMonitoringFeature.h

```diff
@@ -0,0 +1,104 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <map>
+#include <optional>
+#include <vector>
+#include <memory>
+
+#include "Basics/Thread.h"
+#include "Basics/process-utils.h"
+#include "Shell/arangosh.h"
+
+namespace arangodb {
+class ProcessMonitoringFeature;
+
+class ProcessMonitorThread : public arangodb::Thread {
+  friend ProcessMonitoringFeature;
+
+ public:
+  ProcessMonitorThread(application_features::ApplicationServer& server,
+                       ProcessMonitoringFeature* processMonitorFeature)
+      : Thread(server, "ProcessMonitor"),
+        _processMonitorFeature(processMonitorFeature) {}
+  ~ProcessMonitorThread() { shutdown(); }
+
+ protected:
+  ProcessMonitoringFeature* _processMonitorFeature;
```

Comment:
We should use a reference here. That indicates to the reader of the code that there will always be a ProcessMonitorFeature around, and it cannot be a nullptr.
```suggestion
  ProcessMonitoringFeature& _processMonitorFeature;
```